storybookjs/storybook
# Addon-actions: Add depth option for serialization

Please note, I've no idea what i'm doing with typescript. So I expect I'll be told some things.

Issue: https://github.com/storybooks/storybook/issues/6409

## What I did
I threaded through the option "depth" from action through the stringify being called from the channel-postmessage. I also added a base depth to it so that it looks more normal to an end user. For example, depth: 1 corresponds to the top level of whatever object they are passing through.

## How to test
Check in "Button -with text" story of CRA Kitchen sink. When the button is clicked, it should only display the top level of the event object in the Action panel instead of the entire object. This should also be observable in the Chrome profiler as the "stringify" for the event should take much less time.

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.
